### ISSUE TITLE
New Check: Forbid the use of PromoteSingle

### ISSUE THREAD

**Issue by katbyte** [bflad/tfproviderlint#66]
This is no longer valid in .12

**Comment by bflad** [comment]
Certainly matches the `helper/schema.Schema` Go type documentation and should be included with the default checks in `tfproviderlint`

```
    // PromoteSingle originally allowed for a single element to be assigned
    // where a primitive list was expected, but this no longer works from
    // Terraform v0.12 onwards (Terraform Core will require a list to be set
    // regardless of what this is set to) and so only applies to Terraform v0.11
    // and earlier, and so should be used only to retain this functionality
    // for those still using v0.11 with a provider that formerly used this.
    PromoteSingle bool
```

**Comment by bflad** [comment]
New `S034` check will be released in tfproviderlint version 0.10.0, likely later today. 👍
